query.go

```diff
@@ -297,20 +297,19 @@ func (q *query) queryPeer(ctx context.Context, p peer.ID) *queryResult {
 		logger.Debugf("QUERY worker for: %v - not found, and no closer peers.", p)
 	}
 
+	foundCloserPeer := false
 	for _, next := range newPeers {
 		if next.ID == q.dht.self { // don't add self.
 			logger.Debugf("PEERS CLOSER -- worker for: %v found self", p)
 			continue
 		}
 
 		// add their addresses to the dialer's peerstore
-		q.dht.peerstore.AddAddrs(next.ID, next.Addrs, pstore.TempAddrTTL)
-	}
-
-	foundCloserPeer := false
-	for _, np := range newPeers {
-		closer := q.localPeers.Add(np.ID)
-		foundCloserPeer = foundCloserPeer || closer
+		if q.dht.queryPeerFilter(q.dht.host, *next) {
+			q.dht.peerstore.AddAddrs(next.ID, next.Addrs, pstore.TempAddrTTL)
```

Comment:
Is this the correct ordering here, or should we add the addresses to the peerstore and then extract all available addresses and send them through peer filtering?

Reply:
Could you expand on that?

Reply:
This code has the filter take in the host and the AddrInfo. However, it doesn't take into account any addresses currently stored in the peerstore.

It's possible that we might care about other addresses that are already in the peerstore, not just the ones we were told about via the DHT.

Note: I suspect this signatures and logic may also need to change after signed peer records lands.

Reply:
An upside of looking at all the addresses together is that we have more information about the peer we want to judge.

A downside of looking at them all together is that by knowing which addresses came from the DHT as opposed to externally we might have a better idea about if this peer is supposed to be public or private.

We could expose both sets of addresses together, but unless we have something to do with them it's just added complexity for "future-proofing" which seems unhelpful/futile.

@aarshkshah1992 do you think I should just add the addresses to the peerstore and then send them all through the filter?

Reply:
I'd look at all the addresses together. That will give us the best picture.